packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -110,57 +144,195 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
       path: driftStoragePath,
     );
 
-    __analyticsClient = await AnalyticsClient.getInstance(
+    _eventCreator = EventCreator(
+      globalFieldsManager: _eventGlobalFieldsManager,
+      deviceContextInfo: deviceContextInfo,
+    );
+
+    // Retrieve Unique ID
+    final savedFixedEndpointId =
+        await _keyValueStore.read(key: endpointIdStorageKey);
+    final fixedEndpointId = savedFixedEndpointId ?? const Uuid().v1();
+    if (savedFixedEndpointId == null) {
+      await _keyValueStore.write(
+        key: endpointIdStorageKey,
+        value: fixedEndpointId,
+      );
+    }
+
+    final endpoint = PublicEndpoint(
+      effectiveDate: DateTime.now().toUtc().toIso8601String(),
+      demographic: EndpointDemographic(
+        appVersion: deviceContextInfo?.appVersion,
+        locale: deviceContextInfo?.locale,
+        make: deviceContextInfo?.make,
+        model: deviceContextInfo?.model,
+        modelVersion: deviceContextInfo?.modelVersion,
+        platform: deviceContextInfo?.platform?.name,
+        platformVersion: deviceContextInfo?.platformVersion,
+        timezone: deviceContextInfo?.timezone,
+      ),
+      location: EndpointLocation(
+        country: deviceContextInfo?.countryCode,
+      ),
+    );
+
+    /// Retrieve stored GlobalAttributes
```

Comment:
Are you referring to using JsonSerializable to serialize/deserialize types like you did for testEvent in the integrationTest?

If so, is it necessary to do the same for Analytics Dart because for KeyValueStore we're only storing/retrieving a string for endpointId and basic maps for metrics and attributes. 

For the events, we already use the serializer defined for the Pinpoint Event class type.